ssl/ssl3.h

```diff
@@ -627,6 +627,9 @@ typedef struct ssl3_state_st {
      */
     unsigned char *alpn_selected;
     unsigned alpn_selected_len;
+    unsigned char *alpn_proposed;
```

Comment:
The usual place to hide these things is in `s->cert`. You can try and see how e.g. the signature algorithms extensions (`s->cert->peer_sigalgs`) is handled. It'll be ugly, though...